src/components/TableSettingsColumn/TableSettingsColumn.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import {SortableContext, useSortable} from '@dnd-kit/sortable';
+import {CSS} from '@dnd-kit/utilities';
+import {Grip} from '@gravity-ui/icons';
+import {Checkbox, Icon} from '@gravity-ui/uikit';
+import type {Column, Header, Updater, VisibilityState} from '@tanstack/react-table';
+
+import {b} from './TableSettingsColumn.classname';
+import {getIsVisible} from './TableSettingsColumn.utils';
+
+import './TableSettingsColumn.scss';
+
+export const TableSettingsColumn = <TData extends unknown>({
+    column,
+    header,
+    children,
+    visibilityState,
+    onVisibilityToggle,
+}: React.PropsWithChildren<{
+    column: Column<TData, unknown>;
+    header: Header<TData, unknown>;
+    visibilityState: VisibilityState;
+    onVisibilityToggle: (updater: Updater<VisibilityState>) => void;
+}>) => {
+    const innerColumns = column.getLeafColumns();
+    const isVisible = innerColumns.some((innerColumn) =>
+        getIsVisible(innerColumn, visibilityState),
+    );
+
+    const isIndeterminate =
+        isVisible &&
+        innerColumns.some((innerColumn) => !getIsVisible(innerColumn, visibilityState));
+
+    const context = header?.getContext();
+    const columnHeader =
+        typeof column.columnDef.header === 'function'
+            ? column.columnDef.header(context)
+            : column.columnDef.header;
+
+    const toggle = () => {
+        if (innerColumns.length > 1) {
+            const newState = innerColumns.reduce(
+                (acc, innerColumn) => {
+                    const result = {...acc};
+                    result[innerColumn.id] = !isVisible;
+                    return acc;
+                },
+                {} as Record<string, boolean>,
+            );
+            onVisibilityToggle((prevState) => ({...prevState, ...newState}));
+        } else {
+            onVisibilityToggle((prevState) => ({...prevState, [column.id]: !isVisible}));
+        }
+    };
+
+    const renderSpacers = () => {
+        const spacers = [];
+        for (let index = 0; index < column.depth; index++) {
+            spacers.push(<div key={index} className={b('vertical-spacer')} />);
+        }
+        return spacers;
+    };
+
+    const {attributes, listeners, setNodeRef, transform, transition, isDragging} = useSortable({
+        id: column.id,
+    });
+
+    const style = {
+        transform: CSS.Transform.toString(transform),
+        transition,
+    };
```

Comment:
There must be a bug with `transform` style:

https://github.com/user-attachments/assets/db194aa0-3cfa-4b17-8d06-71096bb8d2bf

Reply:
Changed layout and set override for scaleY to 1, now it's looks much better, thx.